alerter/TemperatureThresholdAlerter.java

```diff
@@ -0,0 +1,31 @@
+public class TemperatureThresholdAlerter {
+    public static final String ENV_TYPE = "DEVELOPMENT";
+    static int alertFailureCount = 0;
+
+    static void alertInCelsius(float fahrenheit , NetworkTemperatureAlerter networkTemperatureAlerter) {
+        float celsius = (fahrenheit - 32) * 5 / 9;
+        int returnCode = networkTemperatureAlerter.networkAlertStub(celsius);
+        if (returnCode != 200) {
+            alertFailureCount += 1;
+        }
+    }
+
+    private static NetworkTemperatureAlerter getNetworkAlert() {
+        NetworkTemperatureAlerter networkTemperatureAlerter = null;
+        if (EnvironmentType.DEVELOPMENT.name().equals(ENV_TYPE)) {
```

Comment:
This is a good way of abstracting test implementation from production one i.e. by using interfaces 👍